utils/funcexp/func_regexp.cpp

```diff
@@ -212,39 +215,126 @@ inline bool getBool(rowgroup::Row& row, funcexp::FunctionParm& pm, bool& isNull,
     }
   }
 
-#ifdef __linux__
-  regex_t re;
-
-  regcomp(&re, pattern.c_str(), REG_EXTENDED | REG_NOSUB);
-
-  int res = regexec(&re, expr.c_str(), 0, NULL, 0);
-  regfree(&re);
-
-  if (res == 0)
-    return true;
-  else
-    return false;
-
-#else
-  std::regex pat(pattern.c_str());
-  return std::regex_search(expr.c_str(), pat);
-#endif
+  return RegExpParams{expr, pattern};
 }
-
 }  // namespace
 
 namespace funcexp
 {
+CalpontSystemCatalog::ColType Func_regexp_replace::operationType(FunctionParm& fp,
+                                                                 CalpontSystemCatalog::ColType& resultType)
+{
+  // operation type is not used by this functor
+  return fp[0]->data()->resultType();
+}
+
+CalpontSystemCatalog::ColType Func_regexp_substr::operationType(FunctionParm& fp,
+                                                                CalpontSystemCatalog::ColType& resultType)
+{
+  // operation type is not used by this functor
+  return fp[0]->data()->resultType();
+}
+
+CalpontSystemCatalog::ColType Func_regexp_instr::operationType(FunctionParm& fp,
+                                                               CalpontSystemCatalog::ColType& resultType)
+{
+  // operation type is not used by this functor
+  return fp[0]->data()->resultType();
+}
+
 CalpontSystemCatalog::ColType Func_regexp::operationType(FunctionParm& fp,
                                                          CalpontSystemCatalog::ColType& resultType)
 {
   return resultType;
 }
 
-bool Func_regexp::getBoolVal(rowgroup::Row& row, FunctionParm& pm, bool& isNull,
+using jp = jpcre2::select<char>;
+
+/*
+  returns the string subject with all occurrences of the regular expression pattern replaced by
+  the string replace. If no occurrences are found, then subject is returned as is.
+  https://mariadb.com/kb/en/regexp_replace/
+*/
+std::string Func_regexp_replace::getStrVal(rowgroup::Row& row, FunctionParm& fp, bool& isNull,
+                                           execplan::CalpontSystemCatalog::ColType& ct)
+
+{
+  if (isNull)
```

Comment:
Is this right? We test that something is NULL and exactly what is this something?

Reply:
++